### PR TITLE
Feature/307 bulk node removal

### LINKED ISSUE #307: Support for bulk node removal in job graph
Make it possible to select multiple nodes in the graph and do a batch/group removal of the items.

Note: Keep in mind that there are different types of "things" in the graph. Source tables and components are removed in different ways on the backing job builder.

### PR DESCRIPTION
**Pull request by tomaszguzialek**
Fixes #307

### CODE REVIEW

File: desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphContext.java [hunk 1]
```diff
@@ -88,16 +88,21 @@ public Object getVertex(Point2D p) {
     }
 
     public Object getSelectedVertex() {
```
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphContext.java]: Is this still used anywhere after adding getSelectedVertices?

File: desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphKeyListener.java [hunk 2]
```diff
@@ -49,25 +50,27 @@ public JobGraphKeyListener(JobGraphContext graphContext) {
     public void keyTyped(KeyEvent e) {
         // react to DEL key strokes and delete components that are selected.
         if (e.getKeyChar() == KeyEvent.VK_DELETE) {
-            final Object vertex = _graphContext.getSelectedVertex();
-            logger.debug("Registered typed DEL. Vertex: {}", vertex);
-            if (vertex != null) {
+            final Set<Object> vertices = _graphContext.getSelectedVertices();
+            logger.debug("Registered typed DEL. Vertices: {}", vertices);
+            if ((vertices != null) && (!vertices.isEmpty())) {
```
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphKeyListener.java]: isEmpty() seems unneeded (though harmless)

**Comment by LosD** [comment]
Looks good, but see code comments.

File: desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphKeyListener.java (hunk 2 above)
Comment (tomaszguzialek) [inline desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphKeyListener.java]: I also thought so, but see message of commit no. 2:  Checking if selected vertices set is empty (not only null) :).

File: desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphContext.java (hunk 1 above)
Comment (tomaszguzialek) [inline desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphContext.java]: Does not seem so, removed it now.

File: desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphKeyListener.java (hunk 2 above)
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphKeyListener.java]: But... But... The for loop should not... What? *confused*

**Comment by LosD** [comment]
I have no idea why an empty set would be a problem, but no matter what, the check is not a problem. Merging!

**Pull request merged**